packages/components/src/components/select-wrapper/select-wrapper-dropdown/select-wrapper-dropdown-utils.ts

```diff
@@ -156,6 +154,14 @@ export const getHighlightedOptionMap = (arr: OptionMap[]): OptionMap => arr.find
 export const getValidOptions = (options: OptionMap[]): OptionMap[] =>
   options.filter((item) => !item.hidden && !item.initiallyHidden && !item.disabled);
 
+export const getAmountOfVisibleOptionsAndOptgroups = (options: OptionMap[]): number => {
+  return options.reduce((count, { hidden, initiallyHidden, title }) => {
```

Comment:
Why reduce instead of filter?

Reply:
I had used `filter`before, but it worked only with two iterations, like
```
const visibleItems = options.filter((item) => !item.hidden && !item.initiallyHidden).length;
const itemsWithOptgroups = options.filter((item) => item.title).length;
return visibleItems + itemsWithOptgroups;
  ```
Couldn't find a solution how to combine it in one iteration....

Reply:
No, because we have to find both, all elements which are not "hidden" and additionally all elements with "title".